Error while applying io.flutter.console.FlutterConsoleFilter

## Version information

IntelliJ IDEA `2020.1` • Flutter plugin `io.flutter 45.1.3` • Dart plugin `201.6668.153`

Error getting Flutter sdk information.

## Exception

null

```
java.lang.RuntimeException: java.lang.RuntimeException: Error while applying io.flutter.console.FlutterConsoleFilter@6364b8c9 to 'lib/screens/my_screen.dart:21:48: Error: The method 'getClient' isn't defined for the class 'Repository'.
'
	at com.intellij.execution.impl.AsyncFilterRunner.isQuick(AsyncFilterRunner.java:83)
	at com.intellij.execution.impl.AsyncFilterRunner.highlightHyperlinks(AsyncFilterRunner.java:62)
	at com.intellij.execution.impl.EditorHyperlinkSupport.highlightHyperlinks(EditorHyperlinkSupport.java:287)
	at com.intellij.execution.impl.ConsoleViewImpl.highlightHyperlinksAndFoldings(ConsoleViewImpl.java:987)
	at com.intellij.execution.impl.ConsoleViewImpl.flushDeferredText(ConsoleViewImpl.java:712)
	at com.intellij.execution.impl.ConsoleViewImpl$FlushRunnable.doRun(ConsoleViewImpl.java:1587)
	at com.intellij.execution.impl.ConsoleViewImpl$FlushRunnable.run(ConsoleViewImpl.java:1583)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:232)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:367)
	at com.intellij.util.Alarm$Request.run(Alarm.java:357)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:220)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:310)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:80)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:128)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:184)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:974)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:847)
	at com.intellij.ide.IdeEventQueue.lambda$null$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:739)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:496)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.RuntimeException: Error while applying io.flutter.console.FlutterConsoleFilter@6364b8c9 to 'lib/screens/my_screen.dart:21:48: Error: The method 'getClient' isn't defined for the class 'Repository'.
'
	at com.intellij.execution.filters.CompositeFilter.applyFilter(CompositeFilter.java:82)
	at com.intellij.execution.impl.AsyncFilterRunner$HighlighterJob.analyzeLine(AsyncFilterRunner.java:231)
	at com.intellij.execution.impl.AsyncFilterRunner$HighlighterJob.analyzeNextLine(AsyncFilterRunner.java:220)
	at com.intellij.execution.impl.AsyncFilterRunner$HighlighterJob.access$200(AsyncFilterRunner.java:188)
	at com.intellij.execution.impl.AsyncFilterRunner.runTasks(AsyncFilterRunner.java:148)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.insideReadAction(NonBlockingReadActionImpl.java:485)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$attemptComputation$2(NonBlockingReadActionImpl.java:442)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1106)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:75)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:129)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:625)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:570)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:118)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:75)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.attemptComputation(NonBlockingReadActionImpl.java:450)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$transferToBgThread$0(NonBlockingReadActionImpl.java:383)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:222)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:201)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:210)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:190)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalArgumentException: Invalid range specified: (-1, -1); 
	at com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:252)
	at com.intellij.execution.filters.Filter$ResultItem.<init>(Filter.java:210)
	at com.intellij.execution.filters.Filter$Result.<init>(Filter.java:55)
	at com.intellij.execution.filters.Filter$Result.<init>(Filter.java:48)
	at io.flutter.console.FlutterConsoleFilter.applyFilter(FlutterConsoleFilter.java:191)
	at com.intellij.execution.filters.CompositeFilter.applyFilter(CompositeFilter.java:76)
	... 28 more
```

Probably related:
```
java.lang.RuntimeException: Error while applying io.flutter.console.FlutterConsoleFilter@5a2c58e4 to 'lib/screens/transfer_screen.dart:73:19: Error: Expected 'else' or comma.
'
	at com.intellij.execution.filters.CompositeFilter.applyFilter(CompositeFilter.java:82)
	at com.intellij.execution.impl.AsyncFilterRunner$HighlighterJob.analyzeLine(AsyncFilterRunner.java:231)
	at com.intellij.execution.impl.AsyncFilterRunner$HighlighterJob.analyzeNextLine(AsyncFilterRunner.java:220)
	at com.intellij.execution.impl.AsyncFilterRunner$HighlighterJob.access$200(AsyncFilterRunner.java:188)
	at com.intellij.execution.impl.AsyncFilterRunner.runTasks(AsyncFilterRunner.java:148)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.insideReadAction(NonBlockingReadActionImpl.java:486)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$attemptComputation$2(NonBlockingReadActionImpl.java:443)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1106)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:75)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:129)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:166)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:627)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:572)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:153)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:118)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:75)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.attemptComputation(NonBlockingReadActionImpl.java:451)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$transferToBgThread$0(NonBlockingReadActionImpl.java:384)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:222)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:201)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:210)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:190)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalArgumentException: Invalid range specified: (-1, -1); 
	at com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:252)
	at com.intellij.execution.filters.Filter$ResultItem.<init>(Filter.java:210)
	at com.intellij.execution.filters.Filter$Result.<init>(Filter.java:55)
	at com.intellij.execution.filters.Filter$Result.<init>(Filter.java:48)
	at io.flutter.console.FlutterConsoleFilter.applyFilter(FlutterConsoleFilter.java:191)
	at com.intellij.execution.filters.CompositeFilter.applyFilter(CompositeFilter.java:76)
```

--- a/src/io/flutter/console/FlutterConsoleFilter.java
+++ b/src/io/flutter/console/FlutterConsoleFilter.java
@@ -26,6 +26,9 @@ import io.flutter.sdk.FlutterSdk;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
  * The FlutterConsoleFilter handles link detection in consoles for:
  * <p>
@@ -175,6 +178,27 @@ public class FlutterConsoleFilter implements Filter {
           highlightLength = part.length();
           break;
         }
+      }
+    }
+
+    if (lineStart < 0) {
+      // lib/registerC.dart:104:73: Error: Expected ';' after this.
+      String filePathAndLineNumberExpr = "((?:[^\\/]*\\/)*)(.*):(\\d+):\\d+:";
+      Pattern pattern = Pattern.compile(filePathAndLineNumberExpr);
+      Matcher matcher = pattern.matcher(line);
+      boolean found = matcher.find();
+      if (found) {
+        String path = matcher.group(1) + matcher.group(2);
+        file = fileAtPath(path);
+        if (file == null) {
+          return null;
+        }
+        lineNumber = Integer.parseInt(matcher.group(3));
+        lineStart = entireLength - line.length();
+        highlightLength = path.length();
+      }
+      else {
+        return null;
       }
     }
 

--- a/testSrc/unit/io/flutter/console/FlutterConsoleFilterTest.java
+++ b/testSrc/unit/io/flutter/console/FlutterConsoleFilterTest.java
@@ -49,4 +49,11 @@ public class FlutterConsoleFilterTest {
     Filter.Result link = new FlutterConsoleFilter(fixture.getModule()).applyFilter(line, line.length());
     assertNotNull(link);
   }
+
+  @Test
+  public void checkErrorMessage() {
+    final String line = "test/widget_test.dart:23:18: Error: Expected ';' after this.";
+    final Filter.Result link = new FlutterConsoleFilter(fixture.getModule()).applyFilter(line, line.length());
+    assertNotNull(link);
+  }
 }

